core/src/main/java/org/commonjava/maven/ext/core/impl/RelocationManipulator.java

```diff
@@ -128,13 +161,66 @@ private boolean apply( final Project project, final Model model ) throws Manipul
             result |= updateDependencies( relocations, project.getAllResolvedProfileDependencies( session ).get( profile ) );
 
         }
+
+        result |= updateAllPlugins( model, project, relocations );
+
+        return result;
+    }
+
+    private boolean updateAllPlugins( final Model model, final Project project, final WildcardMap<ProjectVersionRef> relocations ) throws ManipulationException
+    {
+        Map<String, Plugin> pluginMap = getManagedPluginMap( model );
+        List<PluginReference> refs =  findPluginReferences( model, project, pluginMap );
+        boolean result = updatePlugins( relocations,  refs );
+
+        pluginMap = getPluginMap( model );
+        refs = findPluginReferences( model, project, pluginMap );
+
+        result |= updatePlugins( relocations, refs );
+
+        return result;
+    }
+
+    private boolean updatePlugins( final WildcardMap<ProjectVersionRef> relocations, final List<PluginReference> dependencies ) throws ManipulationException
+    {
+        final int size = relocations.size();
+
+        logger.debug( "Got {} relocations for plugins", size );
+
+        boolean result = false;
+
+        for ( int i = 0; i < size; i++ )
+        {
+            final Iterator<PluginReference> it = dependencies.iterator();
+
+            while ( it.hasNext() )
```

Comment:
Could use for loop to stop intellij complaining :-)

Reply:
I am not getting an error/warning here.